### PR TITLE
Add console option to select icon theme

### LINKED ISSUE #40: Set an unicode only icon set
Create the option `--icons` like: 
```sh
lsd --icons=none # Do not display icons
lsd --icons=unicode # Display only unicode icons
lsd --icons=default # Display the default (current) icon set
```
The unicode type is for the user who doesn't want to download a specific font. It should be available with all the classic fonts and terminals.

### PR DESCRIPTION
**Pull request by loewenheim**
I’ve added a console option called `theme` to select an icon set. The possible values are `default` for the current behavior and `unicode` to use only the default unicode icons. Unfortunately, the unicode icons don’t show up correctly on my machine, so I can’t really verify.

This fixes #40.

### CODE REVIEW

File: src/icon.rs [hunk 1]
```diff
@@ -21,10 +22,20 @@ const ICON_SPACE: &str = "  ";
...
 impl Icons {
     pub fn new(theme: Theme) -> Self {
+        let display_icons = theme == Theme::Default || theme == Theme::Unicode;
+        let (icons_by_name, icons_by_extension) = if theme == Theme::Default {
+            (
+                Self::get_default_icons_by_name(),
+                Self::get_default_icons_by_extension(),
+            )
+        } else {
+            (HashMap::new(), HashMap::new())
```
Comment (Peltoche) [inline src/icon.rs]: Here you need to call two new functions `get_unicode_icons_by_name()` and `get_default_icons_by_extension` which will generate two HashMap with the corresponding unicode icons.

Comment (loewenheim) [reply src/icon.rs]: The idea was that if the maps are empty, every call to `get` will fall through to the default case.

File: src/app.rs [hunk 2]
```diff
@@ -32,6 +32,15 @@ pub fn build() -> App<'static, 'static> {
                 .multiple(true)
                 .help("When to print the icons"),
         )
+        .arg(
+            Arg::with_name("theme")
+                .long("theme")
```
[the remark this answers is not in the document]
Comment (meain) [reply src/app.rs]: How about just ``--icons``?

File: src/icon.rs (hunk 1 above)
Comment (loewenheim) [reply src/icon.rs]: What would you say to making the icon set a trait with a single method
```rust
fn get(&self, name: &Name) -> String
```
The unicode icon set would always return the default file or directory icon, the “fancy” icon set would look the icons up in a table. Is that overcomplicating it?

File: src/app.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (loewenheim) [reply src/app.rs]: The problem with `--icons` is that there is already an `--icon` option that decides when to display the icons. i don’t know if renaming the old option is on the table.

[the remark this answers is not in the document]
Comment (meain) [reply src/app.rs]: Oh, yeah. forgot about that. `icons` might cause a confusion and renaming an old option would not be a good idea. Guess `icon-theme` is a pretty good option in this case.

[the remark this answers is not in the document]
Comment (loewenheim) [reply src/app.rs]: `icon-theme` it is, then. Should I change the `default` option to something more descriptive, like `fancy`?

**Comment by loewenheim** [comment]
I don’t know why [this](https://travis-ci.org/Peltoche/lsd/jobs/480480130) CI task fails.
`cargo +beta test --target "x86_64-unknown-linux-gnu" --verbose` works on my computer.

File: src/app.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Peltoche) [reply src/app.rs]: Yep, "fancy" seems cool :)

File: src/icon.rs [hunk 3]
```diff
@@ -22,20 +24,25 @@ const ICON_SPACE: &str = "  ";
...
-        } else {
-            (HashMap::new(), HashMap::new())
-        };
+        let display_icons = theme == Theme::Fancy || theme == Theme::Unicode;
+        let (icons_by_name, icons_by_extension, default_file_icon, default_folder_icon) =
+            if theme == Theme::Fancy {
+                (
+                    Self::get_default_icons_by_name(),
+                    Self::get_default_icons_by_extension(),
+                    "\u{f016}",
```
Comment (Peltoche) [inline src/icon.rs]: Can you add the icon representation in comment please? Like it was made before. It make the code easier to read for the peoples with unicode and/or nerd-font.

File: src/icon.rs [hunk 4]
```diff
@@ -22,20 +24,25 @@ const ICON_SPACE: &str = "  ";
...
+                    "\u{f115}",
+                )
+            } else {
+                (HashMap::new(), HashMap::new(), "\u{1F5CB}", "\u{1F5C1}")
```
Comment (Peltoche) [inline src/icon.rs]: same here. You can lower-case the code points too

File: src/icon.rs [hunk 5]
```diff
@@ -324,12 +331,27 @@ mod test {
...
+        let file_path = tmp_dir.path().join("file");
+        File::create(&file_path).expect("failed to create file");
+        let meta = file_path.metadata().expect("failed to get metas");
+
+        let file_type = FileType::new(&meta, &Permissions::from(&meta));
+        let name = Name::new(&file_path, file_type);
+        let icon = Icons::new(Theme::Unicode);
+        let icon = icon.get(&name);
+
+        assert_eq!(icon, format!("{}{}", "\u{1F5CB}", ICON_SPACE));
```
Comment (Peltoche) [inline src/icon.rs]: same here :)

File: src/icon.rs [hunk 6]
```diff
@@ -338,12 +360,26 @@ mod test {
...
+        let tmp_dir = TempDir::new("test_file_type").expect("failed to create temp dir");
+        let file_path = tmp_dir.path();
+        let meta = file_path.metadata().expect("failed to get metas");
+
+        let file_type = FileType::new(&meta, &Permissions::from(&meta));
+        let name = Name::new(&file_path, file_type);
+        let icon = Icons::new(Theme::Unicode);
+        let icon = icon.get(&name);
+
+        assert_eq!(icon, format!("{}{}", "\u{1F5C1}", ICON_SPACE));
```
Comment (Peltoche) [inline src/icon.rs]: same here

**Comment by Peltoche** [comment]
> I don’t know why [this](https://travis-ci.org/Peltoche/lsd/jobs/480480130) CI task fails.
> `cargo +beta test --target "x86_64-unknown-linux-gnu" --verbose` works on my computer.

I don't know why but sometime a simple restart do the trick. This should be a travis bug I guess...

File: src/icon.rs (hunk 3 above)
Comment (loewenheim) [reply src/icon.rs]: Alright, will do.

**Comment by loewenheim** [comment]
My pleasure :)

**Pull request merged**